File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog/Attributes/AcceptAjaxAttribute.vb
Imports System.Reflection

Namespace Attributes

    Public NotInheritable Class AcceptAjaxAttribute
        Inherits ActionMethodSelectorAttribute

        Public Overrides Function IsValidForRequest(ByVal controllerContext As ControllerContext, ByVal methodInfo As MethodInfo) As Boolean
            If controllerContext Is Nothing Then
                Throw New ArgumentNullException("controllerContext")
            End If
            Return controllerContext.HttpContext.Request.IsAjaxRequest()
        End Function
    End Class

End Namespace

File: AspNetMvcUnleashed/0672329980_source/Chapter20/CS/UnleashedBlog/UnleashedBlog/Attributes/AcceptAjaxAttribute.cs
using System;
using System.Web.Mvc;
using System.Reflection;

namespace UnleashedBlog.Attributes
{

    public sealed class AcceptAjaxAttribute : ActionMethodSelectorAttribute
    {
        public override bool IsValidForRequest(ControllerContext controllerContext, MethodInfo methodInfo)
        {
            if (controllerContext == null)
            {
                throw new ArgumentNullException("controllerContext");
            }
            return controllerContext.HttpContext.Request.IsAjaxRequest();
        }
    }
}